userprefs.php

```diff
@@ -379,7 +380,26 @@ function echo_general_tab($user)
         'credit_name_adhoc',
         null
     );
-    show_blank();
+
+    foreach (get_project_detail_levels() as $level => $label) {
+        // We don't allow users to select level 1 and 4 because:
+        // * Level 1 doesn't include Start Proofreading and will confuse people
+        // * Level 4 includes the page detail which is resource intensive
+        //   and can produce a Very Large Page (>35MB)
+        if ($level == 1 || $level == 4) {
+            continue;
+        }
+
+        $page_detail_options[$level] = sprintf("%s (%d)", $label, $level);
+    }
+    show_preference(
+        _('Default Project Detail Level'),
+        'project_detail',
```

Comment:
Does it make sense to make the setting name a constant to be reference here and in the project page?

Reply:
That's a good idea as that would find problems at "compile" time. We do this (poor) pattern all over the place for user prefs. I think I'd like to be consistent with the others for this PR but think moving to constants for all of these settings is a good idea -- we just need to establish where those constants should live.